From 0901a786d07c42231b31c4bae51ebe9d34ff6990 Mon Sep 17 00:00:00 2001
From: Sunil V L <sunilvl@ventanamicro.com>
Date: Thu, 9 Jun 2022 12:12:13 +0530
Subject: [PATCH 4/5] uefi-sct/SctPkg: Install RISCV_EFI_BOOT_PROTOCOL test
 module

REF: https://bugzilla.tianocore.org/show_bug.cgi?id=3837

Enable installing the RISCV_EFI_BOOT_PROTOCOL test
cases into the SCT directory.

Cc: G Edhaya Chandran <Edhaya.Chandran@arm.com>
Cc: Barton Gao <gaojie@byosoft.com.cn>
Cc: Carolyn Gjertsen <Carolyn.Gjertsen@amd.com>
Cc: Samer El-Haj-Mahmoud <Samer.El-Haj-Mahmoud@arm.com>
Cc: Eric Jin <eric.jin@intel.com>
Cc: Arvin Chen <arvinx.chen@intel.com>
Cc: Supreeth Venkatesh <Supreeth.Venkatesh@amd.com>
Cc: Ard Biesheuvel <ardb@kernel.org>
Cc: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Cc: Abner Chang <abner.chang@hpe.com>

Signed-off-by: Sunil V L <sunilvl@ventanamicro.com>
Reviewed-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Tested-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
---
 uefi-sct/SctPkg/CommonGenFramework.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/uefi-sct/SctPkg/CommonGenFramework.sh b/uefi-sct/SctPkg/CommonGenFramework.sh
index 779a6a44..fb9043b5 100755
--- a/uefi-sct/SctPkg/CommonGenFramework.sh
+++ b/uefi-sct/SctPkg/CommonGenFramework.sh
@@ -222,6 +222,10 @@ then
     cp $ProcessorType/PartitionInfoBBTest.efi                  $Framework/Test/ > NUL
     cp $ProcessorType/UFSDeviceConfigBBTest.efi                $Framework/Test/ > NUL
     cp $ProcessorType/ResetNotificationBBTest.efi              $Framework/Test/ > NUL
+    # Only include RiscVBootProtocolBBTest.efi if the file exists (true on RISCV64)
+    if [ -e $ProcessorType/RiscVBootProtocolBBTest.efi ]; then
+      cp $ProcessorType/RiscVBootProtocolBBTest.efi              $Framework/Test/ > NUL
+    fi
 
     # *********************************************
     # Copy ENTS binary
-- 
2.25.1

